### ISSUE TITLE
Accessibility review

### ISSUE THREAD

**Issue by rianrietveld** [humanmade/hm-platform#54]
As installed on hmn.md version v0.1.0
https://updates.hmn.md/wp-admin/

**Header element**
Not sure about using the `<header class="hm-ek-Header">` element here, as WP core doesn't use any HTML5 landmarks in the admin.
Also this header is put inside the div with role="main" and there is no real header element above that main, which makes this html5 header inside the main a bit messy.
I'd go for a div, like the rest of all the admin screens.

**SVGs overall**
The title attribute is not an existing attribute for an SVG element.
see: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute
use the `<title>` element instead (example below, with the H1).

**SVGs in the menu and toolbar**
The `hm-logo-small` here is decorative, so should be hidden from assistive technology. This can be done by adding role="presentation"
`<svg class="hm-logo-small" role="presentation" viewBox="0 0 547.56 320.92001" height="14.06" width="24">[...]</svg>`

The `hm-logo-small`  is used multiple times in a page. This results in a double ID warning for `<clipPath id="a">`

**SVG in the H1**
The text in the logo is part of the heading text, so should be added as a real title inside the SVG, so it is announced.
add `role="img"` and `aria-labelledby="hm-logo-h1"` to the svg element
and add `<title id="hm-logo-h1">Human Made</title>` inside.
This way the title will be recognized and announced by assistive technology.
```
<h1>
	<svg class="hm-logo-large" role="img" aria-labelledby="hm-logo-h1" viewBox="0 0 876.30664 168.89333" height="168.893" width="876.307">
		<g fill="#d94e3a" transform="matrix(.13333 0 0 -.13333 0 168.893)">
			<title id="hm-logo-h1">Human Made</title>
			[...]
		</g>
	</svg>
	Platform <small class="hm-ek-Header-Version">v0.1.0</small>
</h1>
```
**SVGs in the hm-plugins--plugin box**
Add `role="presentation"`
`<svg role="presentation" width="20px" height="20px" viewBox="0 0 48 48">`

**SVG in the features select**
Add `role="presentation"` and remove the existing title and credits
```
<svg width="20px" height="20px" viewBox="0 0 20 20" version="1.1" role="presentation">
    [...]
</svg>
```
Documentation on a11y SVG:
- [Using ARIA to enhance SVG accessibility](https://developer.paciellogroup.com/blog/2013/12/using-aria-enhance-svg-accessibility/) 
- [Accessible SVGs](https://css-tricks.com/accessible-svgs/)  

**Link texts**
Make the links in the the `hm-plugins--plugin__guides` list unique.
Then the link list will make more sense
Add the name of the plugin as screen reader text to the text like for example:
```
<a class="hm-iframe-open" href="https://docs.humanmade.com/beta/plugins/aws-ses-wp-mail/developer-guide/">
<span class="screen-reader-text">AWS Email </span>Developer Guide
</a>
```

**Features filter**

Select in `hm-plugins--filters__category`:
This works with keyboard, but is totally silent in VoiceOver.
I propose the use of a real select and forget about styling the options, select options are notorious difficult to style & remain accessible.
A good compromise is a styled and accessible selector is used in https://foxland.fi/demo/wordpress-speaks/
Also see my research for Encompass: http://rianrietveld.com/2017/11/21/accessible-custom-styled-form-elements/

Search a category
Add a (screen-reader-text hidden) label and connect it to the input field with a for/id construction like:
```
<label class="screen-reader-text" for="hm-plugins-category">
<input type="text" id="hm-plugins-category" value="" placeholder="Search category">
```
**Plugin status**
`.hm-radio-group `input has `display:none;` this means it will never get focus and is skipped by a keyboard / screen reader.
This construction of styling a radio button is not accessible.

I propose the following (looks probably the same, works for everyone):
Make this a fieldset with a legend, so the Plugin status is announced with the radio buttons.
Hide the input fields with screen-reader-text, so they can get focus.
Add a focus and hover style to the labels, so you can see that the element gets focus (underlined or something else cc @grarighe )
```
<fieldset class="hm-plugins--filters__status hm-radio-group">
	<legend class="screen-reader-text">Plugin Status</legend>
	<input type="radio" id="hm-plugins-status-any" name="active" value="any" class="screen-reader-text"><label for="hm-plugins-status-any">Any</label>
	<input type="radio" id="hm-plugins-status-on" name="active" value="on" class="screen-reader-text"><label for="hm-plugins-status-on">Active</label>
	<input type="radio" id="hm-plugins-status-off" name="active" value="off" class="screen-reader-text"><label for="hm-plugins-status-off">Inactive</label>
</fieldset>
```
**Dynamic search results**
Add an announcement about the found results with wp.a11y.speak
for example:
`wp.a11y.speak( 'Number of plugins found: 16', 'polite' );`
the string can be made translatable with wp_localize_script
Documentation: https://foxland.fi/wordpress-speaks-wp-a11y-speak/

If you add `aria-live-"polite"` to the notice for no plugins found it's also announced, like.
`<p class="notice-large" aria-live-"polite">No plugins found for that combination of filters, try removing your search term or search all categories.</p>`

**Colour contrast**

rules:
Contrast ratio 4.5:1 for normal text
Contrast ratio 3:1 for large text (24px equivalent or 19px equivalent and bold)

Menu:
Menu item HM Platform has a too low contrast:
item is 14 pixels bold
Text: #eff1ef
Background: #d94e3b
Contrast ratio: 3.62:1
Should be 4.5:1

Version number in the header has a too low contrast:
item is 13px, bold
Text: #999999, 
background: #ffffff
Contrast ratio:  2.84 
Should be 4.5:1

Inactive state of the plugin status label a too low contrast:
item is 13px, bold
Text: #999999, 
background: #ffffff
Contrast ratio:  2.84 
Should be 4.5:1

**Heading structure**

Page Enterprise Kit 
H3 in 
The `<h3>The ultimate toolbox for Enterprise</h3>`
should be an H2 (don't skip a heading level)

I'll address the display of the documentation in another issue.
